pkg/component/logging/vali/vali.go

```diff
@@ -0,0 +1,1104 @@
+	}
+	return fmt.Sprintf(`[{"port":%d,"protocol":"TCP"}]`, serviceValiPortNumber)
+}
+
+func fromAllScrapeTargetsAllowedPorts(rbacProxyEnabled bool) string {
+	if rbacProxyEnabled {
+		return fmt.Sprintf(`[{"port":%d,"protocol":"TCP"},{"port":%d,"protocol":"TCP"}]`, serviceValiPortNumber, serviceTelegrafPortNumber)
+	}
+	return fmt.Sprintf(`[{"port":%d,"protocol":"TCP"}]`, serviceValiPortNumber)
+}
+
+func buildValiConfiguration(authEnabled bool) ([]byte, error) {
+	w := bytes.NewBuffer(nil)
+	err := valiConfigTmpl.Execute(w, map[string]interface{}{"AuthEnabled": authEnabled})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create vali configuration: %w", err)
+	}
+	return io.ReadAll(w)
+}
+
+func buildTelegrafConfiguration() ([]byte, error) {
+	w := bytes.NewBuffer(nil)
+	err := telegrafConfigTmpl.Execute(w, map[string]interface{}{"ListenPort": serviceTelegrafPortNumber})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create telegraf configuration: %w", err)
+	}
+	return io.ReadAll(w)
+}
+
+func buildTelegrafStartScript() ([]byte, error) {
+	w := bytes.NewBuffer(nil)
+	err := telegrafStartTmpl.Execute(w, map[string]interface{}{"KubeRBACProxyPort": serviceRBACProxyPortNumber})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create telegraf start script: %w", err)
+	}
+	return io.ReadAll(w)
+}
+
+// TODO: (vlvasilev) adjust the function to use managed resource instead of working directly on the pvc and sts!
+// ResizeOrDeleteValiDataVolumeIfStorageNotTheSame updates the garden Vali PVC if passed storage value is not the same as the current one.
+// Caution: If the passed storage capacity is less than the current one the existing PVC and its PV will be deleted.
+func (v *vali) resizeOrDeleteValiDataVolumeIfStorageNotTheSame(ctx context.Context) error {
+	// Check if we need resizing
+	pvc := &corev1.PersistentVolumeClaim{}
+	if err := v.client.Get(ctx, kubernetesutils.Key(v1beta1constants.GardenNamespace, "vali-vali-0"), pvc); err != nil {
+		return client.IgnoreNotFound(err)
+	}
+
+	storageCmpResult := v.values.Storage.Cmp(*pvc.Spec.Resources.Requests.Storage())
+	if storageCmpResult == 0 {
+		return nil
+	}
+
+	//Annotate managed resource to skip reconciliation.
+	managedResource := &resourcesv1alpha1.ManagedResource{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ManagedResourceControlName,
+			Namespace: v.namespace,
+		},
+	}
+	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, v.client, managedResource, func() error {
+		if managedResource.Annotations == nil {
+			managedResource.Annotations = map[string]string{}
+		}
+		managedResource.Annotations[resourcesv1alpha1.Ignore] = "true"
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	defer func() {
+		_, _ = controllerutils.GetAndCreateOrMergePatch(ctx, v.client, managedResource, func() error {
+			delete(managedResource.Annotations, resourcesv1alpha1.Ignore)
+			return nil
+		})
+	}()
```

Comment:
?

Reply:
What can I say? First I have to annotate the ManagedResource with `igrnore` annotation to scale down the Vali StatefullSet in order to resize the PVC. Then I have to remove the annotation to let GRM do its job. So I put annotation removal at the end of the function. But I ended up in a case (on the kind local setup) where the resizing of the PVC is not enabled. In order to proceed with the reconciliation I had to remove the PV option from the Gardenlet's configuration, but there was no logic to remove the `ignore` annotation because the resizing part was skipped. So I decided to remove the annotation every time the function exits. Eventually, this resulted in a frequent restart of the `vali` pod when resizing is failing, but this is less bad than a blocked reconciliation. 
This third option is to rewrite the logic and not use ManagedResource.